sov-modules/sov-app-template/src/lib.rs

```diff
@@ -100,21 +102,15 @@ where
         batch: Self::Batch,
         sequencer: &[u8],
         _misbehavior_hint: Option<Self::MisbehaviorProof>,
-    ) -> Result<
-        Vec<Vec<sovereign_sdk::stf::Event>>,
-        // Question: Should we return an enum error here? (see TODOs bellow)
-        sovereign_sdk::stf::ConsensusSetUpdate<OpaqueAddress>,
-    > {
+    ) -> anyhow::Result<Vec<Vec<sovereign_sdk::stf::Event>>> {
         let mut batch_workspace = WorkingSet::new(self.current_storage.clone());
         batch_workspace = batch_workspace.to_revertable();
 
-        if self
+        if let Err(e) = self
             .tx_hooks
             .enter_apply_batch(sequencer, &mut batch_workspace)
-            .is_err()
         {
-            // TODO: Enable it after: https://github.com/Sovereign-Labs/sovereign/issues/174
-            //return Ok(Vec::default());
+            anyhow::bail!("Error: enter_apply_batch. {}", e)
```

Comment:
nit: Could we expand the error message here a little. Maybe something like `"Error: The transaction was rejected by the 'enter_apply_batch' hook"`.

Reply:
Fixed